cmd/podman/shared/create.go

```diff
@@ -635,33 +636,46 @@ func ParseCreateOpts(ctx context.Context, c *GenericCLIResults, runtime *libpod.
 	if err != nil {
 		return nil, errors.Wrapf(err, "unable to translate --shm-size")
 	}
-
 	// Verify the additional hosts are in correct format
 	for _, host := range c.StringSlice("add-host") {
 		if _, err := parse.ValidateExtraHost(host); err != nil {
 			return nil, err
 		}
 	}
 
-	// Check for . and dns-search domains
-	if util.StringInSlice(".", c.StringSlice("dns-search")) && len(c.StringSlice("dns-search")) > 1 {
-		return nil, errors.Errorf("cannot pass additional search domains when also specifying '.'")
-	}
-
-	// Check for explicit dns-search domain of ''
-	if c.Changed("dns-search") && len(c.StringSlice("dns-search")) == 0 {
-		return nil, errors.Errorf("'' is not a valid domain")
-	}
-
-	// Validate domains are good
-	for _, dom := range c.StringSlice("dns-search") {
-		if dom == "." {
-			continue
+	var (
+		dnsSearches []string
+		dnsServers  []string
+		dnsOptions  []string
+	)
+	if c.Changed("dns-search") {
+		dnsSearches = c.StringSlice("dns-search")
+		// Check for explicit dns-search domain of ''
+		if len(dnsSearches) == 0 {
+			return nil, errors.Errorf("'' is not a valid domain")
+		}
+		// Check for . and dns-search domains
+		if util.StringInSlice(".", dnsSearches) {
+			if len(dnsSearches) > 1 {
+				return nil, errors.Errorf("cannot pass additional search domains when also specifying '.'")
```

Comment:
We could move the check down to `if dom == "."` in the following loop. Would avoid iterating twice.

Reply:
Fixed